src/bin/ota_plus.rs

```diff
@@ -281,85 +308,181 @@ fn subsubcmd_targets<'a, 'b>() -> App<'a, 'b> {
         .subcommand(
             SubCommand::with_name("remove")
                 .about("Remove a target from the staged metadata")
-                .arg(
-                    Arg::with_name("path")
-                        .help("The target filepath")
-                        .short("p")
-                        .long("path")
-                        .required(true)
-                        .takes_value(true)
-                        .validator(is_target_path),
-                ),
+                .arg(arg_path())
+        )
+        .subcommand(
+            SubCommand::with_name("sign")
+                .about("Sign the targets metadata")
         )
-        .subcommand(SubCommand::with_name("sign").about("Sign the targets metadata"))
-        .subcommand(SubCommand::with_name("push").about("Push the signed targets metadata to the TUF repo"))
+        .subcommand(
+            SubCommand::with_name("push")
+                .about("Push the signed targets metadata to the TUF repo")
+        )
+}
+
+
+fn arg_keyid<'a, 'b>() -> Arg<'a, 'b> {
+    Arg::with_name("keyid")
+        .help("The key ID")
+        .short("i")
+        .long("keyid")
+        .required(true)
+        .takes_value(true)
+        .validator(is_key_id)
+}
+
+fn arg_role<'a, 'b>() -> Arg<'a, 'b> {
+    Arg::with_name("role")
+        .short("r")
+        .long("role")
+        .takes_value(true)
+        .required(true)
+        .possible_values(&["root", "targets", "timestamp", "snapshot"])
+}
+
+fn arg_path<'a, 'b>() -> Arg<'a, 'b> {
+    Arg::with_name("path")
+        .help("The target path")
+        .short("p")
+        .long("path")
+        .required(true)
+        .takes_value(true)
+        .validator(is_target_path)
+}
+
+fn arg_type<'a, 'b>() -> Arg<'a, 'b> {
+    Arg::with_name("type")
+        .short("t")
+        .long("type")
+        .takes_value(true)
+        .required(true)
+        // FIXME(PRO-3849): bouncy castle ed25519 key parsing
+        .possible_values(&["rsa"])
+}
+
+
+fn is_key_id(s: String) -> ::std::result::Result<(), String> {
+    HEXLOWER.decode(s.as_bytes())
+        .map_err(|_| format!("Key ID not hex: {}", s))
+        .and_then(|_| if s.len() != 64 { Err("Key ID should be 64 hex chars".into()) } else { Ok(()) })
+}
+
+fn is_pem_public(s: String) -> ::std::result::Result<(), String> {
+    let mut file = File::open(s).map_err(|e| format!("error opening pem file: {}", e))?;
```

Comment:
We should use `error_chain` for these to preserve the original error in its entirety.

Reply:
Loljk, these are for the the validators for `clap`.